packages/block-library/src/navigation/edit/index.js

```diff
@@ -220,45 +219,80 @@ function Navigation( {
 	// that automatically saves the menu as an entity when changes are made to the inner blocks.
 	const hasUnsavedBlocks = hasUncontrolledInnerBlocks && ! isEntityAvailable;
 
+	const { navigationFallback } = useSelect( ( select ) => {
+		const { getNavigationFallback } = select( coreStore );
+		return {
+			navigationFallback:
+				ref || hasUnsavedBlocks ? null : getNavigationFallback(),
```

Comment:
This aims to avoid fetching fallbacks at all if the block is pre-configured with a `ref` or has uncontrolled inner blocks.